content/posts/aws-rds-upgrade.md

```diff
@@ -0,0 +1,24 @@
+title: Read the Docs Maintenance Window - February 14
+date: 2024-01-31
+description: Upcoming upgrade for PostgreSQL to ensure minimal impact for our users.
+category: Engineering
+tags: aws, rds, postgres, upgrade
+authors: Eric Holscher
+status: draft
+image: /images/posts/aws-rds-upgrade.png
+image_credit:
+
+We wanted to make you aware that on **Friday, February 14 at 2:00pm PST (5:00pm EST, 22:00 UTC)**, Read the Docs will be having a scheduled downtime of approximately 2 hours.
+This will impact both **Read the Docs Community** and **Read the Docs for Business**.
+We expect the downtime to only last for 15-30 minutes, but we are scheduling a longer window to account for any unforeseen issues.
+
+During this maintenance window, documentation will continue to be online but new documentation builds will not trigger and the Read the Docs dashboard will be unavailable.
```

Comment:
> documentation will continue to be online

Is this true? Are we still being able to _read_ data from Postgres? Otherwise, only the documentation cached by CF will be available, right?

Reply:
I believe last time we did this we promoted the read replica so that at least read-only database operations worked.

If we aren't planning on this for this migration, private documentation will not be available.

Reply:
I think we switched the queries over to the read-only DB with a DB router, IIRC.

Reply:
Yeah, likely. I think that was when we had a more complex database router configuration too, the replica might have been the primary database used for auth.

We have a few options we can pick from, worst case we might need a release prior to the window

Reply:
I noted that public docs will continue to work.